libs/common/src/main/java/org/opensearch/common/Numbers.java

```diff
@@ -142,6 +174,31 @@ public static long toLong(String stringValue, boolean coerce) {
         return bigIntegerValue.longValue();
     }
 
+    /** Return the long that {@code stringValue} stores or throws an exception if the
+     *  stored value cannot be converted to a long that stores the exact same
+     *  value and {@code coerce} is false. */
+    public static BigInteger toUnsignedLong(String stringValue, boolean coerce) {
+        final BigInteger bigIntegerValue;
+        try {
+            BigDecimal bigDecimalValue = new BigDecimal(stringValue);
+            if (bigDecimalValue.compareTo(BIGDECIMAL_GREATER_THAN_USIGNED_LONG_MAX_VALUE) >= 0
+                || bigDecimalValue.compareTo(BigDecimal.ZERO) <= 0) {
+                throw new IllegalArgumentException("Value [" + stringValue + "] is out of range for an unsigned long");
+            }
+            bigIntegerValue = coerce ? bigDecimalValue.toBigInteger() : bigDecimalValue.toBigIntegerExact();
+        } catch (ArithmeticException e) {
+            throw new IllegalArgumentException("Value [" + stringValue + "] has a decimal part");
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("For input string: \"" + stringValue + "\"");
+        }
+
+        if (bigIntegerValue.compareTo(MAX_UNSIGNED_LONG_VALUE) > 0 || bigIntegerValue.compareTo(BigInteger.ZERO) < 0) {
```

Comment:
Same about `MIN_UNSIGNED_LONG_VALUE`

Could maybe make a `Numbers.isValidUnsignedLong(BigIngeter)` helper.

Reply:
> Could maybe make a Numbers.isValidUnsignedLong(BigIngeter) helper.

Not sure to be fair, there are only 2 places we use it inside this class